core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -655,6 +655,12 @@ protected void kill() {
         super.kill();
         closeChannel();
         IOUtils.closeQuietly(log);
+
+        try {
+            Util.deleteRecursive(getLogFile().getParentFile());
```

Comment:
To defend against catastrophes from bad future refactorings, I would recommend extracting a protected method for the computer’s log directory, to make it clear that what you are deleting is specific to this computer only.